File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> r;
    int total=brown+yellow,row;
    for(int col=3;col<total+1;col++) {
        row=total/col;
        if((row-2)*(col-2)==yellow)
            break;
    }
    r={row,total/row};
    return r;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total=brown+yellow, row=0, rlt[] = new int[2];
        for(int col=3;col<total+1;col++) {
            row=total/col;
            if((row-2)*(col-2)==yellow)
                break;
        }
        rlt[0]=row;
        rlt[1]=total/row;
        return rlt;
    }
}